feat(codebuild): add ability to customize build status reporting for third-party Git sources

Lilja opened the pull request:
[Documentation](https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.context)

The idea is to be able to customise the Github "check" through the context parameter I linked in the AWS doc.

https://github.com/aws/aws-cdk/pull/19403

skinny85 reviewed (dismissed):
Thanks for the PR @Lilja, but I don't love the current modeling of these capabilities.

Here are the docs of these properties:

```
        /**
         * Specifies the context of the build status CodeBuild sends to the source provider. The usage of this parameter depends on the source provider.
         *
         * - **Bitbucket** - This parameter is used for the `name` parameter in the Bitbucket commit status. For more information, see [build](https://docs.aws.amazon.com/https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/commit/%7Bnode%7D/statuses/build) in the Bitbucket API documentation.
         * - **GitHub/GitHub Enterprise Server** - This parameter is used for the `context` parameter in the GitHub commit status. For more information, see [Create a commit status](https://docs.aws.amazon.com/https://developer.github.com/v3/repos/statuses/#create-a-commit-status) in the GitHub developer guide.
         *
         * @link http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-buildstatusconfig.html#cfn-codebuild-project-buildstatusconfig-context
         */
        readonly context?: string;
        /**
         * Specifies the target url of the build status CodeBuild sends to the source provider. The usage of this parameter depends on the source provider.
         *
         * - **Bitbucket** - This parameter is used for the `url` parameter in the Bitbucket commit status. For more information, see [build](https://docs.aws.amazon.com/https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/commit/%7Bnode%7D/statuses/build) in the Bitbucket API documentation.
         * - **GitHub/GitHub Enterprise Server** - This parameter is used for the `target_url` parameter in the GitHub commit status. For more information, see [Create a commit status](https://docs.aws.amazon.com/https://developer.github.com/v3/repos/statuses/#create-a-commit-status) in the GitHub developer guide.
         *
         * @link http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-buildstatusconfig.html#cfn-codebuild-project-buildstatusconfig-targeturl
         */
        readonly targetUrl?: string;
```

So, what should happen is:

1. The properties used for creating the `BitBucketSource`, `BitBucketSourceProps`, should get a new, optional property called `name`, of type `string`.
2. The properties for creating the `GitHubSource` and `GitHubEnterpriseSource` will need to get a new, optional property, called `context`, of type `string`. (You might have to introduce a common superinterface to share between `GitHubEnterpriseSourceProps` and `GitHubSourceProps`.
3. `ThirdPartyGitSourceProps` should get a new, optional property, called `targetUrl`, of type `string`.
4. You fill the `buildStatusConfig` property of the `SourceProperty` interface that's returned from the `bind()` method of all of these `Source`s (through the `sourceProperty` property of the `SourceConfig` returned from `bind()`).

Let me know if this makes sense!

Thanks,
Adam

skinny85 commented:
Maybe `buildStatusName` and `buildStatusContext` and `buildStatusUrl` are better names for the properties.

Feel free to propose better names, of course 🙂.

Lilja commented:
Hey @skinny85. Thanks for the feedback. I've tried to adress it in my latest commit.

* I've added `GithubContextProps` that is a new interface between github and github enterprise. 
* The `buildStatusConfig` prop is created in `ThirdPartyGitSource`, I didn't exactly understand how you would construct it in the `SourceProvider` since it doesn't have any super config of the more concrete Sources, so it doesn't really know the `context`/`targetUrl`.
* I don't really know how to do the defaults here. It's probably not a good idea to require these to be sent in the props of the concrete github/bitbucket sources, so `buildStatusName: string` should probably be `buildStatusName?: string`. The problem with this is that there is some tooling in this library that requires `@default` documentation of something that is optional. I have no clue what the actual default value is. You can see it in this PR, as name is `AWS CodeBuild us-east-1 (AutoBuildProject89A8053A-LhjRyN9kxr8o)`. So it would seem like the default of `context` would be something like `AWS Codebuild $AWS_REGION ($SOME_SORT_OF_ID)`, but I'm only guessing here. Should we really write a default here when in reality, not supplying `buildStatusConfig` makes the CodeBuild API pick some very opaque default.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: a19a798e47c092389f42f5a92c46a38282fdb255
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=25595072-80e1-492c-98bf-e4ee0aadbc23%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -499,6 +499,15 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
    */
   readonly webhookFilters?: FilterGroup[];
+
+  /**
+ * This parameter is used for the `url`/`target_url` parameter in the Bitbucket/GitHub commit status.
+ * BitBucket: `url`
+ * Github: `target_url`
+ *
+ * @default ???????????
+ */
```
I don't think we need to talk about what exact property this fills for each type. It should be obvious from the name.

Let's change the docs to say something like "The URL that the build will report back to the source provider. Can use built-in CodeBuild variables, like `$AWS_REGION`". Something like that.

Let's also add an `@example`, and make `@default` say something like "- link to the AWS Console for CodeBuild to a particular build execution".

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -537,11 +548,22 @@ abstract class ThirdPartyGitSource extends GitSource {
     if (this.webhookTriggersBatchBuild) {
       project.enableBatchBuilds();
     }
+    const buildStatusConfig = {
+        targetUrl: this.buildStatusUrl
+    }
+    if (typeof superConfig["buildStatusContext"] !== "undefined") {
+        // Github/GitHubEnterprise
+        buildStatusConfig["context"] = superConfig["buildStatusContext"]
+    } else if (typeof superConfig["buildStatusName"] !== "undefined") {
+        // BitBucket
+        buildStatusConfig["context"] = superConfig["buildStatusName"]
+    }
```
Since we need to handle `buildStatusConfig` separately in the two subclasses anyway, let's just remove all of this code.

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -636,10 +658,22 @@ class S3Source extends Source {
+ * wip
+ */
+export interface GithubContextProps extends ThirdPartyGitSourceProps {
+  /**
+   * This parameter is used for the `context` parameter in the GitHub commit status.
+   *
+   * @example 'My build #$CODEBUILD_BUILD_NUMBER'
+   */
+    readonly buildStatusContext: string
```
This time, the property is misaligned:

```suggestion
  /**
   * This parameter is used for the `context` parameter in the GitHub commit status.
   *
   * @example 'My build #$CODEBUILD_BUILD_NUMBER'
   */
  readonly buildStatusContext: string;
```

skinny85 reviewed (dismissed):
We're getting there, thanks for all of your work on this @Lilja!

I hope my review answers all of your questions, if something is still unclear, let me know!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: c8c642808424f5f65f773da84357ced1200b11ca
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9e13685e-ea70-49bc-85db-8f696b825ebb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 1fcd58d7ff1c371c1a43451cde089697d8636948
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=dc30d70f-7408-4198-a5f6-1f4883ad7f33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 82495604353cf1ea54d2fbf6e18ad3e033b62ff9
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bc5350e9-942f-4b30-8d12-4a4b68d45465%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -499,6 +499,14 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
+  /**
+   * This parameter is used for the `url`/`target_url` parameter in the Bitbucket/GitHub commit status.
+   *
+   * @example "The URL that the build will report back to the source provider. Can use built-in CodeBuild variables, like $AWS_REGION"
```
This is a very weird example 😃.

```suggestion
   * @example "$CODEBUILD_PUBLIC_BUILD_URL"
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -499,6 +499,14 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
+   * This parameter is used for the `url`/`target_url` parameter in the Bitbucket/GitHub commit status.
+   *
+   * @example "The URL that the build will report back to the source provider. Can use built-in CodeBuild variables, like $AWS_REGION"
+   * @default "link to the AWS Console for CodeBuild to a particular build execution"
```
No need for the quotes here:

```suggestion
   * @default - link to the AWS Console for CodeBuild to a particular build execution
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -499,6 +499,14 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
+   *
+   * @example "The URL that the build will report back to the source provider. Can use built-in CodeBuild variables, like $AWS_REGION"
+   * @default "link to the AWS Console for CodeBuild to a particular build execution"
+   */
```
Let's also add links to the CodeBuild docs about this functionality:

```suggestion
   *
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.targeturl
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
   */
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -636,10 +646,47 @@ class S3Source extends Source {
+   *
+   * @example "My build #$CODEBUILD_BUILD_NUMBER"
+   * @default "AWS CodeBuild $AWS_REGION ($PROJECT_NAME)"
+   */
```
Let's also add links to the CodeBuild docs about this functionality:

```suggestion
   *
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.context
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
   */
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -768,6 +815,14 @@ export interface BitBucketSourceProps extends ThirdPartyGitSourceProps {
+   *
+   * @example "My build #$CODEBUILD_BUILD_NUMBER"
+   * @default "AWS CodeBuild $AWS_REGION ($PROJECT_NAME)"
+   */
```
Let's also add links to the CodeBuild docs about this functionality:

```suggestion
   *
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.context
   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
   */
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -793,6 +850,10 @@ class BitBucketSource extends ThirdPartyGitSource {
         ...superConfig.sourceProperty,
         location: this.httpsCloneUrl,
+        buildStatusConfig: {
+            context: this.buildStatusName,
+            targetUrl: this.buildStatusUrl,
```
General comment: we use 2 spaces for indentation in the project, not 4.

This needs to be fixed everywhere in the PR (most notably, in the tests).

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+        );
+
+        new codebuild.Project(stack, 'MyProject', {
+          source,
```
Can you inline this `source` variable?

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+    test('github', () => {
+        const context = "My custom CodeBuild worker!"
+        const stack = new cdk.Stack();
+        const source = codebuild.Source.gitHub(
+            { owner: 'awslabs', repo: 'aws-cdk', buildStatusContext: context }
+        );
```
Generally, this is not how we indent these in the project -probably pays to be consistent:

```suggestion
        const source = codebuild.Source.gitHub({
          owner: 'awslabs',
          repo: 'aws-cdk',
          buildStatusContext: context,
        });
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+        new codebuild.Project(stack, 'MyProject', {
+          source,
+        });
+        Template.fromStack(stack).hasResourceProperties(
+            'AWS::CodeBuild::Project', {
+                'Source': {
+                    'buildStatusConfig': {
+                        'context': context,
+                    }
+                }
+            }
+        )
```
Again, this is not consistent with other places in the project:

```suggestion
        Template.fromStack(stack).hasResourceProperties('AWS::CodeBuild::Project', {
          Source: {
            buildStatusConfig: {
              context: context,
            },
          },
        });
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+                }
+            }
+        )
+    })
+    test('github enterprise', () => {
```
Missing semicolon, and empty line between tests:

```suggestion
    });

    test('GitHub Enterprise', () => {
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+        const stack = new cdk.Stack();
+        const source = codebuild.Source.gitHubEnterprise({ 
+            httpsCloneUrl: 'url',
+            buildStatusContext: context
```
Missing trailing comma:

```suggestion
            buildStatusContext: context,
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+            }
+        )
+    })
+    test('bitbucket', () => {
```
```suggestion
    test('BitBucket', () => {
```

skinny85 reviewed (changes requested):
Looks great @Lilja, but you need to fix the formatting to get the PR build to succeed, so we can merge this in!

Lilja commented on packages/@aws-cdk/aws-codebuild/test/codebuild.test.ts:
```diff
@@ -1028,6 +1028,93 @@ describe('secondary sources', () => {
+            'AWS::CodeBuild::Project', {
+                'Source': {
+                    'buildStatusConfig': {
+                        'context': context,
```
Good point, my thought was to have a unit test only having either one of those parameters. The tests above only uses context and name, so my thought was to create a unit test with only targetUrl.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 37df8ac25416417eba77aa472552ff48b560af92
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a009dfa5-34cc-49b6-80c1-a2168e25a0c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 887d8efb7962f130d51ab5ee949572af02339242
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5b263fbf-423e-4a1d-bfa4-67e4e1b05819%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -499,6 +499,17 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
   readonly webhookFilters?: FilterGroup[];
+
+  /**
+   * The URL that the build will report back to the source provider. Can use built-in CodeBuild variables, like $AWS_REGION
```
Multi-line, and full sentences, please:

```suggestion
   * The URL that the build will report back to the source provider.
   * Can use built-in CodeBuild variables, like $AWS_REGION.
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -636,10 +649,50 @@ class S3Source extends Source {
+    return {
+      sourceProperty: {
+        ...superConfig.sourceProperty,
+        buildStatusConfig: {
+          context: this.buildStatusContext,
+          targetUrl: this.buildStatusUrl,
+        },
```
We need to render this property only if at least one of `this.buildStatusContext` or `this.buildStatusUrl` is defined:

```suggestion
        buildStatusConfig: this.buildStatusContext !== undefined || this.buildStatusUrl !== undefined
          ? {
            context: this.buildStatusContext,
            targetUrl: this.buildStatusUrl,
          }
          : undefined,
```

This is why the integration test is failing in the build right now:

```
@aws-cdk/aws-codebuild: Verifying integ.github-webhook-batch.js against integ.github-webhook-batch.expected.json ... CHANGED.
@aws-cdk/aws-codebuild: Resources
@aws-cdk/aws-codebuild: [~] AWS::CodeBuild::Project MyProject39F7B0AE 
@aws-cdk/aws-codebuild:  â””â”€ [~] Source
@aws-cdk/aws-codebuild:      â””â”€ [+] Added: .BuildStatusConfig
@aws-cdk/aws-codebuild: Verifying integ.github.js against integ.github.expected.json ... CHANGED.
@aws-cdk/aws-codebuild: Resources
@aws-cdk/aws-codebuild: [~] AWS::CodeBuild::Project MyProject39F7B0AE 
@aws-cdk/aws-codebuild:  â””â”€ [~] Source
@aws-cdk/aws-codebuild:      â””â”€ [+] Added: .BuildStatusConfig
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -793,6 +859,10 @@ class BitBucketSource extends ThirdPartyGitSource {
       sourceProperty: {
         ...superConfig.sourceProperty,
         location: this.httpsCloneUrl,
+        buildStatusConfig: {
+          context: this.buildStatusName,
+          targetUrl: this.buildStatusUrl,
+        },
```
Same here:

We need to render this property only if at least one of `this.buildStatusContext` or `this.buildStatusUrl` is defined:

```suggestion
        buildStatusConfig: this.buildStatusName !== undefined || this.buildStatusUrl !== undefined
          ? {
            context: this.buildStatusName,
            targetUrl: this.buildStatusUrl,
          }
          : undefined,
```

skinny85 reviewed (changes requested):
We're almost there!

Lilja commented:
Thanks for being so patient with me @skinny85!

Fixed compliance with earlier test cases and rebased the branch. Now, there's only the remaining test case of testing the target_url. Is it interesting or should we remove it completely?

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -636,10 +650,52 @@ class S3Source extends Source {
+ */
+interface CommonGithubSourceProps extends ThirdPartyGitSourceProps {
+  /**
+   * This parameter is used for the `context` parameter in the GitHub commit status.
```
Let's mention the CodeBuild variables here too:

```suggestion
   * This parameter is used for the `context` parameter in the GitHub commit status.
   * Can use built-in CodeBuild variables, like $AWS_REGION.
```

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -636,10 +650,52 @@ class S3Source extends Source {
+    this.buildStatusContext = props.buildStatusContext;
+  }
+
+  public bind(_scope: CoreConstruct, project: IProject): SourceConfig {
```
Should be `scope`, not `_scope`.

skinny85 commented on packages/@aws-cdk/aws-codebuild/lib/source.ts:
```diff
@@ -768,6 +824,17 @@ export interface BitBucketSourceProps extends ThirdPartyGitSourceProps {
   readonly repo: string;
+
+  /**
+   * This parameter is used for the `name` parameter in the Bitbucket commit status.
```
Here too:

```suggestion
   * This parameter is used for the `name` parameter in the Bitbucket commit status.
   * Can use built-in CodeBuild variables, like $AWS_REGION.
```

skinny85 reviewed (commented):
Last 3 cosmetic changes, and this will be merged in!

> Fixed compliance with earlier test cases and rebased the branch. Now, there's only the remaining test case of testing the target_url. Is it interesting or should we remove it completely?

Nah, we're good. Leave it as-is.

> Thanks for being so patient with me @skinny85!

My pleasure, thanks for all of your hard work in contributing this feature!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d28d345cd4dc7f7aa75fefbbb2ffc09f4c937eb0
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6378c221-fe0f-45ff-bf7f-72f2976e5ed1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 85947310aeea341e816ee09b64d990993dfec481
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5db14e25-7492-42cb-bf7a-20c4e1b84942%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 52972d4086d60a972a1404097a0a8db03f882891
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e47b1875-574b-43f3-aa00-37261de3ad47%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 reviewed (approved):
Looks great @Lilja!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: fe25f02d3791b3449d6dbcacc7ce7466828fee50
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2795942a-e008-471d-869b-d8a46e123164%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
